assembl/static2/js/app/pages/multiColumnsAdmin.jsx

```diff
@@ -1,9 +1,97 @@
 import React from 'react';
+import { compose, graphql } from 'react-apollo';
+import { connect } from 'react-redux';
+import { get } from '../utils/routeMap';
+import ExportSection from '../components/administration/exportSection';
+import Navbar from '../components/administration/navbar';
+import DiscussionPreferenceLanguage from '../graphql/DiscussionPreferenceLanguage.graphql';
 
 class MultiColumnsAdmin extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      exportLocale: null,
+      refetching: false,
+      translate: false
+    };
+  }
+
+  componentDidMount() {
+    this.props.router.setRouteLeaveHook(this.props.route, this.routerWillLeave);
+  }
+
+  componentWillUnmount() {
+    this.props.router.setRouteLeaveHook(this.props.route, null);
+  }
+
+  handleExportLocaleChange = (locale) => {
```

Comment:
I would bind these functions in the constructor of this class. 
eg. 
```
this.handleExportLocaleChange = this.handleExportLocaleChange.bind(this);
```

https://medium.com/shoutem/react-to-bind-or-not-to-bind-7bf58327e22a

Reply:
But the point of using an arrow function here is to avoid having to bind the function

Reply:
Okay I didn't know. I think we have been using arrow functions all over the project

Reply:
So I made the modifications. But I just found this https://github.com/facebook/react/issues/9851
apparently it's pretty much the same thing

Reply:
@ayyazdaniaryan class properties are transformed with babel (click on the Try button in the example https://babeljs.io/docs/plugins/transform-class-properties/) to an efficient syntax. You can have perf issue with binding in the render method yes because it will create new functions, but class properties are completely different here. We prefer the class properties syntax in the project because of flow.